repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletResponseWrapper.scala

```diff
@@ -465,6 +498,13 @@ class HttpServletResponseWrapper(originalResponse: HttpServletResponse, headerMo
       originalResponse.flushBuffer()
     }
 
+    if (sentError) {
```

Comment:
Ah. So the thing is, since both `flushBuffer` and `sendError` commit the response, both flags should not be set at the same time. However, if a user calls `uncommit` between calling those methods, you could end up in that state. I spent a bunch of time trying to figure out all of the possibilities, but apparently I missed that one.

Since both of these calls commit the underlying response, I'll just use an if/else. I'll give priority to `sendError` since it can be explicitly unset (and should be if the intent is not to send an error but just to flush the buffer).